src/node/NodeWebSocketTransport.ts

```diff
@@ -16,9 +16,14 @@
 import NodeWebSocket from 'ws';
 import { ConnectionTransport } from '../common/ConnectionTransport.js';
 import { packageVersion } from '../generated/version.js';
+import { promises as dns } from 'dns';
 
 export class NodeWebSocketTransport implements ConnectionTransport {
-  static create(url: string): Promise<NodeWebSocketTransport> {
+  static async create(urlString: string): Promise<NodeWebSocketTransport> {
+    const url = new URL(urlString);
+    const { address } = await dns.lookup(url.hostname, { verbatim: false });
```

Comment:
why do we need an explicit DNS lookup? please link to the issue/documentation describing it (I assume it's needed by Node 18). Why does not the NodeWebSocket lib not handle that?

Reply:
I've already commented on this. Are you able to see my comments?

Reply:
could you please include that as a comment in the code here to make sure we don't loose the context? (I still have not understood the issue though)

Reply:
Done.